### PR TITLE
Generate test session during tests

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Description

Generate test JWE cookie during tests to avoid broken tests.

### CODE REVIEW

File: test/fixture/sessionEncryption.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { JWK, JWE } = require('jose');
```
Comment (joshcanhelp) [inline test/fixture/sessionEncryption.js]: Adapted from existing cookie encryption:

https://github.com/auth0/express-openid-connect/blob/master/lib/appSession.js

**Comment by joshcanhelp** [comment]
We'll need to merge in #48 before these tests pass.

File: test/appSession.tests.js [hunk 2]
```diff
@@ -54,19 +55,11 @@ describe('appSession', function() {
...
-    const result = appSessionMw(thisReq, res, next);
-
-    it('should call next', function() {
-      assert.ok(result);
-    });
+    const thisReq = {get: () => 'identity=' + sessionEncryption.encrypted};
 
     it('should set the identity on req', function() {
+      const result = appSessionMw(thisReq, res, next);
+      assert.ok(result);
```
Comment (lbalmaceda) [inline test/appSession.tests.js]: If the line above is async (see `next` is being passed), how can you ensure the result will be set by when this line is reached? Unless I'm missing something

File: test/fixture/sessionEncryption.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+const hkdf = require('futoin-hkdf');
+
+const deriveKey = (secret) => hkdf(secret, 32, { info: 'JWE CEK', hash: 'SHA-256' });
+const epoch = () => Date.now() / 1000 | 0;
```
Comment (lbalmaceda) [inline test/fixture/sessionEncryption.js]: Is this working? Or the pipe should be an OR? Also, I'd add parentheses for clarifying the scope.

```suggestion
const epoch = () => (Date.now() / 1000) || 0;
```

File: test/fixture/sessionEncryption.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const key = JWK.asKey(deriveKey('__test_secret__'));
+const payload = JSON.stringify({sub: '__test_sub__'});
+
+const encryptOpts = {
+  alg: 'dir',
+  enc: 'A256GCM',
+  zip: 'DEF',
+  uat: epoch(),
+  iat: epoch(),
+  exp: epoch() + ( 7 * 24 * 60 * 60 )
```
Comment (lbalmaceda) [inline test/fixture/sessionEncryption.js]: Given that epoch is a function, is there any chance that there might be a slight diff between these 3 different calls? and if that affects the token, wouldn't it be better to call the function once before defining the `encryptOpts` object?

Another nit one, I'd move `7 * 24 * 60 * 60` to a constant and give it a name, or at least add a comment giving the quick result e.g. `// 1 week in secs`

File: test/fixture/sessionEncryption.js (hunk 3 above)
Comment (joshcanhelp) [reply test/fixture/sessionEncryption.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#.7c_%28Bitwise_OR%29

Comment (joshcanhelp) [reply test/fixture/sessionEncryption.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators#.7c_%28Bitwise_OR%29

File: test/appSession.tests.js (hunk 2 above)
Comment (joshcanhelp) [reply test/appSession.tests.js]: The `next()` there isn't a good indication of whether that function is async or not. You would see `await` in front if it was.

**Pull request merged**